dix/rpcbuf.c

```diff
@@ -0,0 +1,187 @@
+/* SPDX-License-Identifier: MIT OR X11
+ *
+ * Copyright © 2024 Enrico Weigelt, metux IT consult <info@metux.net>
+ */
+#include <dix-config.h>
+
+#include <stddef.h>
+
+#include "dix/rpcbuf_priv.h"
+
+Bool x_rpcbuf_makeroom(struct x_rpcbuf *rpcbuf, size_t needed)
+{
+    /* not allocated yet ? */
+    if (!rpcbuf->buffer) {
+        if (!(rpcbuf->buffer = calloc(1, XLIBRE_RPCBUF_CHUNK_SIZE)))
+            return FALSE;
+        rpcbuf->size = XLIBRE_RPCBUF_CHUNK_SIZE;
+        rpcbuf->pos = 0;
+    }
+
+    /* still enough space */
+    if (rpcbuf->size > rpcbuf->pos + needed)
+        return TRUE;
+
+    const size_t newsize = ((needed / XLIBRE_RPCBUF_CHUNK_SIZE) + 1)
+                                * XLIBRE_RPCBUF_CHUNK_SIZE;
+
+    char *newbuf = realloc(rpcbuf->buffer, newsize);
+    if (!newbuf)
+        return FALSE;
+    memset(newbuf + rpcbuf->size, 0, newsize - rpcbuf->size);
+    free(rpcbuf->buffer);
```

Comment:
This free should be removed now that is does realloc

Reply:
Indeed. thanks.